pkg/kubeapiserver/admission/initializer.go

```diff
@@ -67,6 +75,14 @@ func (i pluginInitializer) Initialize(plugin admission.Interface) {
 		wants.SetInternalClientSet(i.internalClient)
 	}
 
+	if wants, ok := plugin.(WantsToRun); ok {
```

Comment:
> The internal client has to be always set before the Run is invoked. So there must be at least some guaranteed ordering. I would invoke the Run ASAP.

Since derek called out the ordering problem, I considered the openshift case. We're likely to have a plugininitializer chain that includes this plugininitializer *and* an openshift specific one.  Our admission plugins won't be able to `Run` until after both are completed.